File: 0001. Two Sum/swift/Solution.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var diff: [Int:Int] = [:];

        for i in 0 ..< nums.count {
            let value = nums[i]
            let found = target - value

            guard let index = diff[found] else {
                diff[value] = i
                continue
            }
            return [index, i]
        }
        return [-1, -1]
    }
}

File: 0001. Two Sum/c++/solution.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> diff;

        for (int i = 0; i < nums.size(); i++) {
            int value = nums[i];
            int found = target - value;

            if (diff.count(found)) {
                return {diff[found], i};
            }
            diff[value] = i;
        }
        return {-1, -1};
    }
};
